homeassistant/components/alarmdecoder/entity.py

```diff
@@ -0,0 +1,24 @@
+"""Support for AlarmDecoder-based alarm control panels entity."""
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .const import (
+    DOMAIN,
+)
+
+
+class AlarmDecoderEntity(Entity):
+    """Define a base AlarmDecoder entity."""
+
+    _attr_has_entity_name = True
+
+    def __init__(self, client):
+        """Initialize the alarm decoder entity."""
+        self._client = client
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, self._client.serial_number)},
+            manufacturer="NuTech",
+            serial_number=self._client.serial_number,
+            sw_version=self._client.version_number,
+        )
```

Comment:
I think actually we don't need to make `_client` accessible anymore, as it was only necessary for the `device_info`.
```suggestion
        self._attr_device_info = DeviceInfo(
            identifiers={(DOMAIN, client.serial_number)},
            manufacturer="NuTech",
            serial_number=client.serial_number,
            sw_version=client.version_number,
        )
```

Reply:
It's used for more than just the registration.  I need adext functionality across child entities.  As a reference, see alarm_control_panel which calls a couple functions on it.

Reply:
Indeed!
We should still use the local variable directly inside attr_device_info:
```suggestion
        self._client = client
        self._attr_device_info = DeviceInfo(
            identifiers={(DOMAIN, client.serial_number)},
            manufacturer="NuTech",
            serial_number=client.serial_number,
            sw_version=client.version_number,
        )
```

Reply:
Yup that seems fair & more readable.